### DISCUSSION TITLE
Starting the SO VM takes a very long time

### DISCUSSION THREAD

**Discussion by iqworks** [Security-Onion-Solutions/securityonion#8907] [Unsupported Versions] [answered]
Hi, 
    I am using :
    SO version 22.04.
    I am trying to use SO on my windows 10 laptop, I don’t use the cloud.
    I used securityonion-2.3.160-20220829.iso
    I used Centos 7
    For nodes, I just have laptops, TV’s and a printer.
    I am using a dell laptop and windows 10 version 10.0.19044 build 19044

   For the past 2 months SO VMs have started as usual bringing everything to OK status. But for the past 6 days its take a long time (5 to 10  minutes or so) to bring all services to an OK status. 

  Also, if i leave the VM process for a few minutes, the cursor disappears and the process just sits there or I get this type of message :
![image](https://user-images.githubusercontent.com/5569263/195411504-58bb14cd-3c9b-4a3e-b5e3-a1823ef22796.png)
This means that I have to run  sudo so-status every 15 - 20 seconds, and sometimes I get all OK's and I can get into my SOC. 

  I restarted my laptop and power on my VM, it seemed to go a little faster. When the problem happened again, I restarted my laptop and it still took to long to get my OK’s. 

  I stopped my anti virus process, still no change. 

  I reinstalled VMware, still no change. 

  Once I changed the memory from 16 to 32 GB, this seem to help, but I don’t think this is a good idea going forward. 

  I created another VM with the same ISO, still no change. 

I also did a defragmentation and compaction, still no change. 

  Not sure what to try next?

    Thanks for your advice and suggestions

**Comment by iqworks** [comment]
I called my anti virus support. They saw that my paging was not configured correctly. These are the steps they went through : 
To change paging :
1 – settings and enter performance in the search. 
![image](https://user-images.githubusercontent.com/5569263/195728110-ad05208a-9493-499d-bfbe-597a4d69e387.png)
2 – select adjust the appearance and performance of windows.
![image](https://user-images.githubusercontent.com/5569263/195728135-943d73b1-a3bc-47e9-886e-9c766c6bdf3e.png) 
3 – select advanced and select change. 
![image](https://user-images.githubusercontent.com/5569263/195728160-6b1a55c0-8d91-409e-a5fb-c53ab9c38d70.png)
4 – now I changed my paging from
![image](https://user-images.githubusercontent.com/5569263/195728191-fb930733-dad2-4be7-ba27-5e6d60f13ba5.png) 
To  
![image](https://user-images.githubusercontent.com/5569263/195728209-bb370657-7459-461a-bfaf-8f3d536e9d31.png)
5 -  I then loaded my usual daily windows files and ran my SO VM. it ran quicker and I left it for a while and no errors, just needed to do ctrl to get the cursor back. Looked like the problem was fixed, but the next day the problem still exist.

**Comment by dougburks** [comment, marked as answer]
This most likely means that your VM is under-powered. Please review the Hardware Requirements:
https://docs.securityonion.net/en/2.3/hardware.html

**Reply by iqworks** [reply]
thanks Doug. The thing is that in my post i mentioned that it worked for the last two months? It has been having this issue since 10/06/2022. It finally works after several tries, but it is very slow. 
  I ran this command "sudo so-docker-refresh" and it looks like it sped things up, but the two SO VMs i ran it on now say elastalert is MISSING. Not sure if that command had anything to do with that?
  I tried finding out how to re-install elastalert. I did see your post from another community member about "reverting my elastalert config about reverting back my elastalert and restarting it". I think reverting back means to go back to the original copy of elastalert. I couldnt find the elastalert config?
thanks.

**Reply by dougburks** [reply]
For your virtual machine, what are the RAM, CPU, and storage specs?

For your physical host, what are the RAM, CPU, and storage specs? Is the storage a mechanical drive or SSD?

**Reply by iqworks** [reply]
thanks for your time on this Doug, i think this is all you need, if not let me know. Also, is there some sort of timeout limit? It seens like it freezes when i dont interact with the VM screen?
![image](https://user-images.githubusercontent.com/5569263/196455850-a20c60c8-0b0c-4a10-b24b-573cddbbd6df.png)
![image](https://user-images.githubusercontent.com/5569263/196455906-b741bbc4-f6a2-4656-a628-da8d05e92846.png)

For my VM
![image](https://user-images.githubusercontent.com/5569263/196456022-16f0a166-ab5a-45cd-b0cc-c38478a8b927.png)

Drives
![image](https://user-images.githubusercontent.com/5569263/196456269-24f67d25-a591-48c2-986a-9db6fcfadfc3.png)

Current disk space
![image](https://user-images.githubusercontent.com/5569263/196456548-d47d06d3-2ee1-4d60-9978-b14520e7c1c2.png)

Running the VM. It froze here 
![image](https://user-images.githubusercontent.com/5569263/196459183-890e71cf-6147-46f6-8948-d9b030f29a83.png)

Here is a look at my task manager.
![image](https://user-images.githubusercontent.com/5569263/196466449-bb4dd34e-d3aa-4210-a13e-ae99a40b6057.png)

I restarted and walked away without pressing enter on the password. When I came back 5 minutes later, it was stuck.
![image](https://user-images.githubusercontent.com/5569263/196463163-133f37b0-0113-4e46-b4d2-b7d8c9fc3c8b.png)

it also gives a beep when it gets stuck and you leave it there in its stuck state. 

thanks again for any advice or suggestions

**Reply by dougburks** [reply]
Your Windows Task Manager screenshot shows that your laptop is also running SQL Server and this could be interfering with performance. If you're going to run Security Onion as a VM, you should ensure that it has sufficient resources allocated to it and that there is no resource contention with other processes. For best results, we recommend installing Security Onion on its own dedicated hardware:
https://docs.securityonion.net/en/2.3/best-practices.html#installation

Your VMware screenshot shows that your Security Onion VM has 14.9GB RAM. This is low for most use cases:
https://docs.securityonion.net/en/2.3/hardware.html#standalone-deployments

Your VMware screenshot shows that your Security Onion VM is running from D: which is a hard disk drive. We typically recommend SSD for best performance:
https://docs.securityonion.net/en/2.3/hardware.html#elastic-stack

It's also possible that your D: drive is experiencing errors which could result in the freezing that you're seeing. You can check your Windows log and run diagnostics on your D: drive, but that is beyond the scope of this discussion forum.

In summary, you'll have the best experience if you install Security Onion on dedicated hardware as described in our Hardware Requirements and Best Practices pages:
https://docs.securityonion.net/en/2.3/hardware.html#virtualization
https://docs.securityonion.net/en/2.3/best-practices.html#installation

**Reply by iqworks** [reply]
Thanks Doug. I do have a machine that will have 16GB. I  will read your links and research your comments. 
thanks again for taking the time.

**Reply by dougburks** [reply]
Please keep in mind that 16GB RAM may still be low depending on how you configure the system and how much data you intend to process. For most production deployments, 16GB is considered a bare minimum and more RAM will provide better performance. Please review the hardware requirements thoroughly for your intended configuration:
https://docs.securityonion.net/en/2.3/hardware.html#standalone-deployments

**Reply by iqworks** [reply]
thanks again Doug, I am reading this link right now. Unfortunatly, the spare laptop I have maxes out at 16GB. But I will be researching and reading all the links you sent me. 
thanks again for your time

**Reply by iqworks** [reply]
I took your advice and eliminated some processes services i didnt really need running. And Dell took over my laptop on thursday and cleared my cache and temporary files. I let SO run over the weekend and today, Monday it is still running, not freezing and not using as much CPU. i think this helped tremendously!
I also got my dedicated laptop with 16GB and created a new VM, installed SYSMON and winlogbeat. I know 16GB is the minimum, but it seems to be working fine. 
thanks for the advice!
